### PR TITLE
Adding documentation website

### PR DESCRIPTION
**Pull request by TOsmanov**
#### Contains

This PR contains a website created using an extended version of Hugo ([hugo documentation](https://gohugo.io/documentation/)) and a docsy theme ([docsy documentation](https://www.docsy.dev/docs/)) .

- The main documentation page has original design. 
- All documentation from "docs" folder move to "site/content/en/docs/" directory.
- Document user_manual.md divided to few sections.
- Added submodule the docsy theme.
- Temporary website files added to the .gitignore file
- Added site/README.md - instruction for local start website
- Updated diagrams using [mermaid](https://www.docsy.dev/docs/adding-content/lookandfeel/#diagrams-with-mermaid)

#### Local run

For local start documentation site [install hugo](https://github.com/gohugoio/hugo/releases/tag/v0.86.0) and add a path to "hugo" in the "Path" environment variable. After this you can run:

```
git clone https://github.com/TOsmanov/datumaro.git -b add-website
cd datumaro
git submodule update --init --recursive
cd site 
hugo server
```

#### Site preview 
https://tosmanov.github.io/datumaro/docs/

#### Documentation structure
```
Documentation
│   contribution.md
│   design.md
│   developer_manual.md
│   getting_started.md
│   release_notes.md
│
├───formats
│       cifar.md
│       cityscapes.md
│       coco.md
│       image_zip.md
│       kitti.md
│       kitti_raw.md
│       mnist.md
│       open_images.md
│       pascal_voc.md
│       sly_pointcloud.md
│       yolo.md
│
├───plugins
│       openvino_plugin.md
│
└───user-manual
    │   command-line-workflow.md
    │   data-formats.md
    │   extending.md
    │   installation.md
    │   interfaces.md
    │   links.md
    │   supported-formats.md
    │
    └───command-reference
            add-and-remove-data.md
            compare-projects.md
            convert-datasets.md
            create-project.md
            explain-inference.md
            export-project.md
            filter-project.md
            get-project-info.md
            get-project-statistics.md
            import-project.md
            merge-projects.md
            register-model.md
            run-model.md
            transform-project.md
            update-project.md
            validate-project-annotations.md
```

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by TOsmanov** [comment]
https://tosmanov.github.io/datumaro/docs/

**Comment by zhiltsov-max** [comment]
1. image

The button in the header looks extra on a documentation website, at least now.

2. Let's remove tags, I don't see any use for them now. They need to be more focused.
3. Release note -> Release note**s**
4. The topics in the Developer manual section are split into sub-topics: 
  image
  But Getting Started has all the info in a single topic: 
  image
  Can we have uniform layout of sections? I think, I like the single-article style more, but the navigation bar on the left side, just inside the other sections.

**Comment by TOsmanov** [comment]
> 1. image
> 
> The button in the header looks extra on a documentation website, at least now.
> 
> 1. Let's remove tags, I don't see any use for them now. They need to be more focused.
> 2. Release note -> Release note**s**
> 3. The topics in the Developer Manual section are split into sub-topics:
>    image
>    But Getting Started has all the info in a single topic:
>    image
>    Can we have uniform layout of sections? I think, I like the single-article style more, but the navigation bar on the left side, just inside the other sections.
> 4. Developer Guide -> Developer Manual

To display sub-topics in the left menu, the document must be divided into separate files, and the content on the right is generated by headings automatically. I divided large sections such as "user manual" and "developer guide" into several files, this separation simplifies co-authoring (if the file is one, then conflicts may arise when working together), as well as text on several pages is easier to read. 
"Getting started" seemed like a small partition to me so I didn't split it into multiple files. 
Maybe it's better to split Getting started?

**Comment by zhiltsov-max** [comment]
> Maybe it's better to split Getting started?

No, I'd choose joining developer manual sections instead.

**Comment by TOsmanov** [comment]
@zhiltsov-max  
Should I delete docs/cli_design.mm?

**Comment by zhiltsov-max** [comment]
Yes. I thought, the whole directory will be removed.

File: site/config.toml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+[[menu.main]]
+    name = "GitHub"
+    weight = 50
+    url = "https://github.com/openvinotoolkit/datumaro"
+    pre = "<i class='fab fa-github'></i>"
+
+[services]
+[services.googleAnalytics]
+# Comment out the next line to disable GA tracking. Also disables the feature described in [params.ui.feedback].
+id = "UA-00000000-0"
```
Comment (azhavoro) [inline site/config.toml:64]: It's better to comment this line

**Comment by nmanovic** [comment]
@TOsmanov , I cannot find description for some annotation formats (https://tosmanov.github.io/datumaro/docs/formats/). For example, ICDAR, Market-1501, LFW, WiderFace, etc. Could you please check?

**Comment by TOsmanov** [comment]
> @TOsmanov , I cannot find description for some annotation formats (https://tosmanov.github.io/datumaro/docs/formats/). For example, ICDAR, Market-1501, LFW, WiderFace, etc. Could you please check?

I took the format descriptions from https://github.com/openvinotoolkit/datumaro/tree/develop/docs/formats for the formats you are talking about there is no user manual

**Review by zhiltsov-max — APPROVED** [review]
Looks good enough for the initial version, from my point of view. We need to start updating the #393.

**Comment by zhiltsov-max** [comment]
Please fix the problems, so the PR could be merged.

**Comment by zhiltsov-max** [comment]
Could you also add a line in changelog?

**Comment by zhiltsov-max** [comment]
How do we enable this for the repository?

**Pull request merged**